backend/LeetWars.Builder/Services/XmlTestResultParserService.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                nsManager.AddNamespace("ns", xmlDoc.DocumentElement.NamespaceURI);
+            }
+
+            XmlNode? countersNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:ResultSummary/ns:Counters", nsManager);
+
+            XmlNode? resultSummaryNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:ResultSummary", nsManager);
+
+            XmlNode? startTimeNode = xmlDoc.SelectSingleNode("/ns:TestRun/ns:Times", nsManager);
+
+            XmlNodeList? unitTestResultNodes = xmlDoc.SelectNodes("/ns:TestRun/ns:Results/ns:UnitTestResult", nsManager);
+
+            if (countersNode == null || resultSummaryNode == null || startTimeNode == null)
+            {
+                throw new ArgumentException("Incorrect xml test file supplied, cannot find needed data");
+            }
+
+            string? passed = countersNode.Attributes?["passed"]?.Value;
+
+            string? failed = countersNode.Attributes?["failed"]?.Value;
+
+            string? outcome = resultSummaryNode.Attributes?["outcome"]?.Value;
+
+            string? startTimeString = startTimeNode.Attributes?["start"]?.Value;
+
+            string? finishTimeString = startTimeNode.Attributes?["finish"]?.Value;
+
+            if (passed == null || failed == null || outcome == null || startTimeString == null || finishTimeString == null || unitTestResultNodes == null)
+            {
+                throw new ArgumentException("Incorrect xml test file supplied, cannot find needed data");
+            }
+
+            ICollection<Test> tests = new List<Test>();
+
+            foreach (XmlNode testNode in unitTestResultNodes)
+            {     
+                tests.Add(ParseSingleCSharpTest(testNode, nsManager));
+            }
+
+            DateTime startTime = DateTime.Parse(startTimeString, CultureInfo.InvariantCulture);
+
+            DateTime finishTime = DateTime.Parse(finishTimeString, CultureInfo.InvariantCulture);
+
+            TimeSpan totalTime = finishTime - startTime;
+
+
+            var testResult = new TestsOutput(outcome == "Passed", totalTime.ToString(), tests, int.Parse(failed), int.Parse(passed));
+
+            return testResult;
+        }
+
+        public TestsOutput ParseJSTestResult(string xmlTestResult)
+        {
+            var xmlDoc = new XmlDocument();
+
+            xmlDoc.LoadXml(xmlTestResult);
+
+            XmlNode? testSuits = xmlDoc.SelectSingleNode("//testsuites") ?? throw new ArgumentException("Incorrect xml test file supplied, cannot find needed data");
+
+            string? duration = testSuits.Attributes?["time"]?.Value;
+
+            string? testsCount = testSuits.Attributes?["tests"]?.Value;
+
+            string? failed = testSuits.Attributes?["failures"]?.Value;
+
+            if(duration == null || testsCount == null || failed == null) 
+            {
+                throw new ArgumentException("Incorrect xml test file supplied, cannot find needed data");
+            }
```

Comment:
I believe you can create just some kind of generic method and reuse it in all similar places

Reply:
Created